File: Algorithms/Seminars/Seminar_2/BinarySearch.java
package Seminar_2;

public class BinarySearch {
    // Функция для поиска элемента в отсортированном массиве
    public static int search(int[] sortedArray, int target) {
        int left = 0; // Начало массива
        int right = sortedArray.length - 1; // Конец массива

        // Пока начало массива меньше или равно его концу
        while (left <= right) {
            int middle = (left + right) / 2; // Находим средний элемент

            // Если средний элемент равен искомому значению, возвращаем его индекс
            if (sortedArray[middle] == target) {
                return middle;
            }
            // Если средний элемент меньше искомого значения, ищем в правой половине массива
            if (sortedArray[middle] < target) {
                left = middle + 1;
            } else {
                // Иначе ищем в левой половине массива
                right = middle - 1;
            }
        }
        return -1; // Если элемент не найден, возвращаем -1
    }
}

File: Paradigms/HW6/binary_search.py
def binary_search(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

# Пример использования:
arr = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]
target = 11
index = binary_search(arr, target)

if index != -1:
    print(f"Элемент {target} найден в массиве, его индекс: {index}")
else:
    print(f"Элемент {target} не найден в массиве")
